### DISCUSSION TITLE
feature request: influxdb: influxql: choose database in query editor

### DISCUSSION THREAD

**Discussion by keeks5** [grafana/grafana#38527] [Observability]
Hi,

I described my concern here. https://community.grafana.com/t/possible-to-use-query-builder-for-other-influxdb-schemas/52329

Is is possible to develop option to define the database name for the query builder interface under query options?(or somewhere else) So we could use the builder instead the editor when user wants to access different database that is defined in the data source configuration. 
![image](https://user-images.githubusercontent.com/88368478/130747860-ca023404-e05a-4157-a905-d6c7425baecf.png)

Reason: If users have multiple InfluxDBs configured and there are also several databases in InfluxDB. Then the data sources configuration could have hundred of entries and it is hard to maintain.
The query builder is often more suitable for the users. Rather to use the editor, for editor you need to know exactly what you need. Also, for some people who are not so familiar with the syntax etc writing the query in editor is difficult and time consuming.

Br,
K

**Comment by mckn** [comment]
@grafana/observability-squad

**Comment by gabor** [comment]
hi @keeks5 thanks for the information, as you know this is currently not possible in the query-editor (you can do it in the raw query mode). i marked this discussion as a feature-request, so we can track this and discuss possible approaches.

**Comment by keeks5** [comment]
Hello,

But what are the options here? How to achieve the solution?

Br,
K

**Reply by gabor** [reply]
hi, sorry, i mean, in the query builder interface it is currently not possible.
but it should be possible to access a different database using the raw query mode.

**Reply by keeks5** [reply]
Yes that I am aware of, I meant about my original request. That user could define the desired database in the options etc so they could use the builder. Who could help me with that? It would need some development.

**Comment by gelicia** [comment]
Hello, as you may have heard, we are [transitioning away from using discussions to discuss feature requests](https://github.com/grafana/grafana/discussions/73424). Due to the age and number of responses to this discussion, we are deciding to close it. If this is something you would like to see in Grafana, feel free to open an [issue](https://github.com/grafana/grafana/issues) so the discussion can continue. Thank you!
